[Trivial] Update comments referencing main.cpp

CryptAxe opened the pull request:
Update comments and documentation referencing main.cpp.

CryptAxe committed:
[Trivial] Update comments referencing main.cpp

laanwj commented:
Yes makes sense https://github.com/bitcoin/bitcoin/pull/9763/commits/5ca5db250c23a6181f80ad858f4ded226baf8712

paveljanik commented:
ACK https://github.com/bitcoin/bitcoin/pull/9763/commits/5ca5db250c23a6181f80ad858f4ded226baf8712

fanquake commented:
utACK 5ca5db2

jnewbery commented on src/test/DoS_tests.cpp:
```diff
@@ -23,7 +23,7 @@
 #include <boost/test/unit_test.hpp>
 
-// Tests this internal-to-main.cpp method:
+// Tests this internal-to-net_processing.cpp method:
```
May as well correct the entire comment while you're doing this:

`// Tests these internal net_processing.cpp methods:`

jnewbery commented on src/txmempool.cpp:
```diff
@@ -395,7 +395,7 @@ bool CTxMemPool::addUnchecked(const uint256& hash, const CTxMemPoolEntry &entry,
     NotifyEntryAdded(entry.GetSharedTx());
     // Add to memory pool without checking anything.
-    // Used by main.cpp AcceptToMemoryPool(), which DOES do
+    // Used by validation.cpp AcceptToMemoryPool(), which DOES do
```
Why not remove the name of the source file entirely?

CryptAxe replied:
It's useful to know where the transaction is being checked before it is added using this function. Otherwise I guess you could use grep to figure out where AcceptToMemoryPool() went but why not just say where it is?

jnewbery replied:
grep, ack, tag files, ... there are many ways to find where the AcceptToMemoryPool() definition is. Naming the source file doesn't add any clarity here, and as you say adds cost to future refactors.

CryptAxe replied:
Fair enough :+1:

CryptAxe committed:
Update internal-to-net_processing.cpp comment

CryptAxe committed:
Remove name of file from comment

Remove name of file where AcceptToMemoryPool() is defined from comment. This will prevent anyone from having to update the comment again with future refactoring.

jnewbery commented:
utACK

fanquake commented:
re-utACK 17dc03c

laanwj commented:
Squashed https://github.com/bitcoin/bitcoin/pull/9763/commits/5ca5db250c23a6181f80ad858f4ded226baf8712 https://github.com/bitcoin/bitcoin/pull/9763/commits/6878c0655a295877f54c31e2b5758925108e70f5 https://github.com/bitcoin/bitcoin/pull/9763/commits/17dc03c839c2b7141cd14b6ff51c18a232d2977e
into 00e623d0b8cd1c261a2671aa8ab221a0474bc7c8 and merged via 8743320.
